internal/tofu/marks.go

```diff
@@ -59,10 +43,45 @@ func marksEqual(a, b []cty.PathValueMarks) bool {
 	return true
 }
 
-func copyMarksFromValue(dst, src cty.Value) cty.Value {
-	_, pvm := src.UnmarkDeepWithPaths()
-	if len(pvm) == 0 {
-		return dst
+// combinePathValueMarks will combine the marks from two sets of marks with paths, ensuring that we don't duplicate marks
+// for the same path, but instead combine the marks for the same path
+// This ensures that we don't lose user marks when combining 2 different sets of marks for the same path
+func combinePathValueMarks(marks []cty.PathValueMarks, other []cty.PathValueMarks) []cty.PathValueMarks {
+	// skip some work if we don't have any marks in either of the lists
+	if len(marks) == 0 {
+		return other
+	}
+	if len(other) == 0 {
+		return marks
 	}
-	return dst.MarkWithPaths(pvm)
+
+	combined := make([]cty.PathValueMarks, 0, len(marks))
+	// construct the initial set of marks
+	combined = append(combined, marks...)
+
+	// check if we've already inserted this by looping over and calling .Equals().
+	// This isn't so nice but there is no nice comparison for cty.PathValueMarks
+	// so we have to do it this way
+	for _, mark := range other {
+		exists := false
+		for i, existing := range combined {
+			if mark.Path.Equals(existing.Path) {
+				// if we found a matching path, we should combine the marks and update the existing item
+				dupe := existing
+				for k, v := range mark.Marks {
+					dupe.Marks[k] = v
+				}
+				combined[i] = dupe
+				exists = true
+				break
```

Comment:
Are we sure `marks` slice contains unique paths?

Reply:
it's a map under the hood, so in theory it should be unique 👍